test/xspec-variable.xspec

```diff
@@ -5,29 +5,212 @@
+      <t:call function="my:foo">
+        <t:param name="elem"><foo/></t:param>
+        <t:param name="p" select="0"/>
+        <t:param name="textnode" select="$myspace1"/>
+      </t:call>
+      <t:expect label="String with correct space" test="$x:result//string/string() = 'abc&#x09;&#x0A;&#x0D;&#x20;  def'"/>
+    </t:scenario>
+    <t:scenario label="Text defined in code to be tested">
+      <t:variable name="myspace2" as="text()" select="$my:myspace"/>
+      <t:call function="my:foo">
+        <t:param name="elem"><foo/></t:param>
+        <t:param name="p" select="0"/>
+        <t:param name="textnode" select="$myspace2"/>
+      </t:call>
+      <t:expect label="String with correct space" test="$x:result//string/string() = 'abc&#x09;&#x0A;&#x0D;&#x20;  def'"/>
+    </t:scenario>
+    <t:scenario label="Text defined in XML file">
+      <t:variable name="myspace3" as="text()" href="xspec-variable.xml" select="//myspace/text()"/>
+      <t:call function="my:foo">
+        <t:param name="elem"><foo/></t:param>
+        <t:param name="p" select="0"/>
+        <t:param name="textnode" select="$myspace3"/>
+      </t:call>
+      <t:expect label="String with correct space" test="$x:result//string/string() = 'abc&#x09;&#x0A;&#x0D;&#x20;  def'"/>
+    </t:scenario>
+  </t:scenario>
+
+  <t:scenario label="Nonsense variable definition in pending scenario does not cause build failure">
+    <t:scenario label="Pending scenario 1" pending="skip on purpose">
+      <t:variable name="nonsense1" select="$nonexistent"/>
+      <t:call function="my:square">
+        <t:param select="$nonsense1"/>
+      </t:call>
+      <t:expect label="the result" test="0"/>
+    </t:scenario>
+    <t:pending label="skip on purpose">
+      <t:scenario label="Pending scenario 2">
+        <t:variable name="nonsense2" href="nonexistent.xml"/>
+        <t:call function="my:square">
+          <t:param select="$nonsense2"/>
+        </t:call>
+        <t:expect label="the result" test="0"/>
+      </t:scenario>
+    </t:pending>
+    <t:scenario label="Nonpending scenario (not needed, except to work around issue #446)">
+      <t:call function="my:square">
+        <t:param select="$test-3/@num" as="attribute(num)"/>
+      </t:call>
+      <t:expect label="the result" select="$test-3/@value cast as xs:integer"/>
+    </t:scenario>
+  </t:scenario>
+
+  <t:scenario shared="yes" label="SHARED variables">
+    <!-- When referencing this scenario via t:like, make sure $str is already defined. -->
+    <t:variable name="bracketed" select="concat('[',$str,']')" as="xs:string"/>
+  </t:scenario>
+
+  <t:scenario label="Interaction of variables and reuse of scenarios">
+    <t:scenario label="Scenario 1">
+      <t:call function="my:square">
+        <t:param select="xs:integer(0)"/>
+      </t:call>
+      <t:variable name="str" select="'abc'" as="xs:string"/>
+      <t:like label="SHARED variables"/>
```

Comment:
The `like` element doesn't seem to be documented. I found one test in the Schematron area but I didn't see tests for usage of `like` for XSLT or XQuery tests. I think the feature is useful and it appears to work. I would like to see it get documented eventually. But let me know if I should remove the usage of it in this test because it's premature.

Reply:
So I submitted #476 for testing `x:like`. It's not comprehensive but better than nothing.

Reply:
`x:like` documentation TODO submitted as #477.

Although not documented, I think it's ok to use `x:like` in this test file, because `x:like` has been available since the very initial commit in 2008.